modules/openapi-generator/src/main/java/org/openapitools/codegen/DefaultGenerator.java

```diff
@@ -1112,6 +1141,31 @@ private void processOperation(String resourcePath, String httpMethod, Operation
         if (operation.getParameters() != null) {
             for (Parameter parameter : operation.getParameters()) {
                 operationParameters.add(generateParameterId(parameter));
+
+                if ((parameter instanceof QueryParameter || "query".equalsIgnoreCase(parameter.getIn()))
+                        && parameter.getSchema() != null && parameter.getSchema().get$ref() != null
+                        && Parameter.StyleEnum.DEEPOBJECT != parameter.getStyle()) {
+
+                    Optional<CodegenModel> matchedModel = allModels.stream()
+                        .map(ModelMap::getModel)
+                        .filter(codegenModel -> codegenModel != null
+                            && codegenModel.getName() != null
+                            && codegenModel.getName().equalsIgnoreCase(parameter.getName())).findFirst();
+
+                    if (matchedModel.isPresent()) {
+                        CodegenModel model = matchedModel.get();
+
+                        Schema schema = parameter.getSchema();
+                        parameter.set$ref(schema.get$ref());
+                        schema.set$ref(null);
+                        schema.setType(model.getDataType().toLowerCase(Locale.ROOT));
+
+                        Map<String, Schema> properties = mapProperties(model);
+                        if(!properties.isEmpty()) {
+                            schema.setProperties(properties);
+                        }
+                    }
+                }
```

Comment:
Suggestions: please add some comments explaining what this code block does.

Reply:
Done